### ISSUE TITLE
CASL Angular: Function calls are not supported...

### ISSUE THREAD

**Issue by wickedsp1d3r** [stalniy/casl#270]
Hello,
Recently, when I try to extract i18n messages from source code with command...

```
ng xi18n myproject --output-path src/locale && ngx-extractor -i src/**/*.ts -o src/locale/messages.xlf
```

...the process interrupts and an error is thrown:

```
ERROR in Error during template compile of 'AppModule'
    Function calls are not supported in decorators but 'AbilityModule' was called.
Unexpected value 'undefined' imported by the module 'AppModule in <path>/src/app/app.module.ts'
Error during template compile of 'AppModule'
    Function calls are not supported in decorators but 'AbilityModule' was called.
```

I have `AbilityModule` included in `AppModule` just like the documentation [shows](https://github.com/stalniy/casl/tree/master/packages/casl-angular#1-including-module).
I have Angular v9.0.3 running.

Any help is appreciated.

**Comment by stalniy** [comment]
Hello,

Did you enable Ivy? According to https://medium.com/angular-in-depth/solving-aot-error-in-ngrx-function-calls-are-not-supported-in-decorators-5c337381457a this error is not thrown if you use Ivy

**Comment by stalniy** [comment]
If you cannot enable Ivy don’t use `forRoot` method and just use `AbilityModule`, and create provider for `Ability` manually  according to the article.

**Comment by wickedsp1d3r** [comment]
Thank you for the answer.

I have Ivy enabled.
Removing `forRoot` and use only `AbilityModule` leads to another error:
`Unexpected value 'undefined' exported by the module`

**Comment by stalniy** [comment]
Do you use the latest version of @casl/angular?

**Comment by wickedsp1d3r** [comment]
Yes, I am using the latest version.

**Comment by stalniy** [comment]
OK, I've tried to migrate [casl/angular](https://github.com/stalniy/casl-angular-example) to 9.x and got similar errors during `ng update` process. But when I run `ng serve` and `ng build`, everything is fine - app works as expected and without errors

**Comment by stalniy** [comment]
Ohhh... I hate angular with all its overengeenering... I spent the whole day to understand where the issue is.

Unexpectedly angular team doesn't recommend to use Ivy for library authors because something was not settled down in its API - https://indepth.dev/the-angular-ivy-guide-for-library-authors/ . Instead they suggest to publish View Engine compatible code and use `ngcc` in the app to magically rewrite libraries to support Ivy... But obviously `ngcc` expects code to be built in a specific way or buy specific tools (like `ng` or `ngc`), otherwise it cannot detect the places which needs to be rewritten and shows extremely strange errors in the terminal

The long story short. It's angular. It always was like this starting from 1.x. Now it's fixed.

Published in `@casl/angular@3.0.6`
